db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/pre_deployment.rb

```diff
@@ -1,93 +1,53 @@
-require 'net/ssh'
-
 LOCAL_BOOK = 'local_book.yaml'.freeze
-REPO_URL = "https://copr.fedorainfracloud.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo".freeze
-
-def handle_rhel_subscriptions(commands)
-  system "scp -o 'StrictHostKeyChecking no' rhel_subscribe_inventory.yaml " \
-    "#{$evm.root['user']}@#{$evm.root['deployment_master']}:~/"
-  rhel_subscribe_cmd = "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
-                       "/usr/share/ansible/openshift-ansible/rhel_subscribe_inventory.yaml"
-  commands.push("sudo mv ~/rhel_subscribe_inventory.yaml /usr/share/ansible/openshift-ansible/", rhel_subscribe_cmd)
-end
-
-def ssh_exec!(ssh, command)
-  stdout_data, stderr_data = "", ""
-  exit_code, exit_signal = nil, nil
-
-  ssh.open_channel do |channel|
-    channel.exec("ssh -A -o 'StrictHostKeyChecking no' -t -t #{$evm.root['user']}@#{$evm.root['deployment_master']} " \
-                 + command) do |_, success|
-      raise StandardError, "Command \"#{command}\" was unable to execute" unless success
-
-      channel.on_data do |_, data|
-        stdout_data += data
-      end
-
-      channel.on_extended_data do |_, _, data|
-        stderr_data += data
-      end
-
-      channel.on_request("exit-status") do |_, data|
-        exit_code = data.read_long
-      end
+REPO_URL   = "https://copr.fedorainfracloud.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo".freeze
 
-      channel.on_request("exit-signal") do |_, data|
-        exit_signal = data.read_long
-      end
-    end
+def pre_deployment(user, deployment_master, deployment)
+  $evm.log(:info, "********************** #{$evm.root['ae_state']} ***************************")
+  deployment.perform_scp(deployment_master, user, "inventory.yaml", "inventory.yaml")
+  release = deployment.perform_agent_commands(deployment_master, user, ["sudo cat /etc/redhat-release"])[:stdout]
+  commands = ['sudo yum install -y ansible-1.9.4',
+              'sudo yum install -y openshift-ansible openshift-ansible-playbooks pyOpenSSL',
+              "sudo mv ~/inventory.yaml /usr/share/ansible/openshift-ansible/",
+              "sudo yum install -y atomic-openshift-utils"]
+  if release.include?("CentOS")
+    commands.unshift("sudo yum install epel-release -y",
+                     "sudo curl -o /etc/yum.repos.d/maxamillion-origin-next-epel-7.repo #{REPO_URL}",
+                     "sudo yum install centos-release-paas-common centos-release-openshift-origin -y")
```

Comment:
This function is now much cleaner :clap:

I'd just perform() these commands here and move the `commands = [...]` above into perform() call below.
You'd lose single-connection optimization but IMO you do enough of individual perform calls that it doesn't matter.
Your call (pun intended ;-)